includes/admin/class-wp-job-manager-promoted-jobs.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Promoted_Jobs.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles promoted jobs functionality.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Promoted_Jobs {
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var self
+	 * @since  $$next-version$$
+	 */
+	private static $instance = null;
+
+	/**
+	 * Allows for accessing single instance of class. Class should only be constructed once per call.
+	 *
+	 * @since  $$next-version$$
+	 * @static
+	 * @return self Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'manage_edit-job_listing_columns', [ $this, 'columns' ] );
+		add_action( 'manage_job_listing_posts_custom_column', [ $this, 'custom_columns' ], 2 );
+		add_action( 'admin_notices', [ $this, 'my_admin_notice' ] );
```

Comment:
I wonder if another hook more generic, like `admin_footer` would make more sense here.

Also, I imagine this class is something generic for many things related to promoted jobs, right? If yes, I'd suggest to use some more specific name for these methods, just to avoid confusion when the class grows.

Reply:
I like that idea!